sixtyfps_compiler/passes/ensure_window.rs

```diff
@@ -0,0 +1,99 @@
+/* LICENSE BEGIN
+    This file is part of the SixtyFPS Project -- https://sixtyfps.io
+    Copyright (c) 2020 Olivier Goffart <olivier.goffart@sixtyfps.io>
+    Copyright (c) 2020 Simon Hausmann <simon.hausmann@sixtyfps.io>
+
+    SPDX-License-Identifier: GPL-3.0-only
+    This file is also available under commercial licensing terms.
+    Please contact info@sixtyfps.io for more information.
+LICENSE END */
+//! Make sure that the top level element of the component is always a Window
+
+use crate::expression_tree::Expression;
+use crate::langtype::Type;
+use crate::namedreference::NamedReference;
+use crate::object_tree::{Component, Element};
+use crate::typeregister::TypeRegister;
+use std::cell::RefCell;
+use std::collections::HashSet;
+use std::rc::Rc;
+
+pub fn ensure_window(component: &Rc<Component>, type_register: &TypeRegister) {
```

Comment:
I had to read this function a few times :) and it makes sense to me. I’m just wondering about one detail: properties declared in new_root seem like they would not be access I through the API (interpreter/C++/Rust) since they remain in new_root. Am I missing something perhaps?

Reply:
no, properties are not moved, so they remain in what becomes the window

Reply:
Ah yes, of course they remain in win_elem